source/reference/indexes.txt

```diff
@@ -1,278 +1,213 @@
+needs to scan. If your application is repeatedly running queries
+on certain fields, you can create an index on those fields to improve query
+performance.
 
-.. code-block:: ruby
+The following sections in this guide describe how to declare and create different
+types of indexes using {+odm+}. The examples use the ``Restaurant`` model, which 
+maps to the ``restaurants`` collection in the ``sample_restaurants`` database. 
+To learn how to connect to this database
+and collection using {+odm+}, see the :ref:`mongoid-quick-start-rails` or 
+:ref:`mongoid-quick-start-sinatra` guides.
 
-    class Person
-      include Mongoid::Document
-      field :ssn
+Declare and Create an Index
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-      index({ ssn: 1 }, { unique: true, name: "ssn_index" })
-    end
+When using {+odm+}, you can declare your index using the ``index`` macro and 
+then create it using the ``create_indexes`` macro.
 
-You can define indexes on embedded document fields as well:
+The following code example shows how to declare and create an ascending index 
+named ``cuisine_index`` on the ``cuisine`` field in the ``Restaurant`` class:
 
-.. code-block:: ruby
+.. literalinclude:: /includes/reference/indexes.rb
+  :language: ruby
+  :emphasize-lines: 8, 11
+  :start-after: start create index
+  :end-before: end create index
 
-    class Person
-      include Mongoid::Document
-      embeds_many :addresses
-      index "addresses.street" => 1
-    end
+The ``index`` macro defines the index you want to create and the ``create_indexes``
+macro creates the index in the ``restaurants`` collection. 
 
-You can index on multiple fields and provide direction:
+When defining an index, the first hash object contains the field you want to
+index and its direction. ``1`` represents an ascending index, ``-1`` represents a
+descending index. The second hash object contains index options. To learn more
+about index options, see the :ref:`mongoid-indexes-api-documentation` section.
 
-.. code-block:: ruby
+Aliases and Declaring Indexes
++++++++++++++++++++++++++++++
 
-    class Person
-      include Mongoid::Document
-      field :first_name
-      field :last_name
+You can use aliased field names in index definitions. For example, the following
+example creates a field on the ``b`` field, which is an alias of the ``borough``
+field:
 
-      index({ first_name: 1, last_name: 1 }, { unique: true })
-    end
+.. literalinclude:: /includes/reference/indexes.rb
+  :language: ruby
+  :start-after: start create alias index
+  :end-before: end create alias index
 
-Indexes can be sparse:
+Create an Index on Embedded Document Fields
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. code-block:: ruby
+You can define an index on embedded document fields. The following code example
+shows how to declare an ascending index on the ``street`` field in the
+``Restaurant`` class where ``street`` is a field embedded within the ``address``
+field.
```

Comment:
[s] This description was hard for me to parse, I wonder if something like this works:
```suggestion
You can define an index on embedded document fields. The following code example
shows how to declare an ascending index on the ``street`` field, which is embedded within the ``address``field in the ``Restaurant`` class.
```

Or break it up:
```suggestion
You can define an index on embedded document fields. The following code example
shows how to declare an ascending index on the ``street`` field in the
``Restaurant`` class. The ``street`` field is embedded within the ``address``
field.
```